src/pages/invoices/common/hooks/useInvoiceColumns.tsx

```diff
@@ -0,0 +1,422 @@
+    },
+    {
+      column: 'client_country',
+      id: 'client_id',
+      label: t('client_country'),
+      format: (value, invoice) =>
+        invoice.client?.country_id &&
+        resolveCountry(invoice.client?.country_id)?.name,
+    },
+    {
+      column: 'client_state',
+      id: 'client_id',
+      label: t('client_state'),
+      format: (value, invoice) => invoice.client?.state,
+    },
+    {
+      column: 'contact_email',
+      id: 'client_id',
+      label: t('contact_email'),
+      format: (value, invoice) =>
+        invoice.client &&
+        invoice.client.contacts.length > 0 && (
+          <CopyToClipboard text={invoice.client?.contacts[0].email} />
+        ),
+    },
+    {
+      column: 'contact_name',
+      id: 'client_id',
+      label: t('contact_email'),
+      format: (value, invoice) =>
+        invoice.client &&
+        invoice.client.contacts.length > 0 &&
+        `${invoice.client?.contacts[0].first_name} ${invoice.client?.contacts[0].last_name}`,
+    },
+    {
+      column: 'created_at',
+      id: 'created_at',
+      label: t('created_at'),
+      format: (value) => date(value, dateFormat),
+    },
+    {
+      column: 'created_by',
+      id: 'user_id',
+      label: t('created_by'),
+      format: (value) => '', // Need to fetch the user with the invoice.
+    },
+    {
+      column: 'custom1',
+      id: 'custom_value1',
+      label: customField(company?.custom_fields.invoice1).label(),
+    },
+    {
+      column: 'custom2',
+      id: 'custom_value2',
+      label: customField(company?.custom_fields.invoice2).label(),
+    },
+    {
+      column: 'custom3',
+      id: 'custom_value3',
+      label: customField(company?.custom_fields.invoice3).label(),
+    },
+    {
+      column: 'custom4',
+      id: 'custom_value4',
+      label: customField(company?.custom_fields.invoice4).label(),
+    },
+    {
+      column: 'discount',
+      id: 'discount',
+      label: t('discount'),
+      format: (value, invoice) =>
+        formatMoney(
+          value,
+          invoice.client?.country_id || company?.settings.country_id,
+          invoice.client?.settings.currency_id || company?.settings.currency_id
+        ),
+    },
+    {
+      column: 'documents',
+      id: 'documents',
+      label: t('documents'),
+      format: (value, invoice) => invoice.documents.length,
+    },
+    {
+      column: 'entity_state',
+      id: 'id',
+      label: t('entity_state'),
+      format: (value) => '', // Need to calculate manually
+    },
+    {
+      column: 'exchange_rate',
+      id: 'exchange_rate',
+      label: t('exchange_rate'),
+    },
+    {
+      column: 'is_deleted',
+      id: 'is_deleted',
+      label: t('is_deleted'),
+      format: (value, invoice) => (invoice.is_deleted ? t('Yes') : t('No')),
+    },
+    {
+      column: 'is_viewed',
+      id: 'id',
+      label: t('is_viewed'),
+      format: (value, invoice) => '', // Need to calculate manually
```

Comment:
@turbo124 How's this calculated?

Reply:
On the invitation there will be a viewed_at column, if it is not null, then it has been viewed.